src/service/data_products/taxa_count.py

```diff
@@ -55,12 +65,20 @@ class Ranks(BaseModel):
 _FLD_KEY = "key"
 
 
-# TODO DATAPROD add auth and allow admins to override load_ver
-@_ROUTER.get(f"/collections/{{collection_id}}/{ID}/ranks", response_model=Ranks)
-async def get_ranks(r: Request, collection_id: str = PATH_VALIDATOR_COLLECTION_ID):
+@_ROUTER.get(
+    f"/collections/{{collection_id}}/{ID}/ranks",
+    response_model=Ranks,
+    description="Get the taxonomy ranks, in rank order, for the taxa counts.\n\n "
+        + "Authentication is not required unless overriding the load version, in which case "
+        + "service administration permissions are required.")
+async def get_ranks(
+    r: Request,
+    collection_id: str = PATH_VALIDATOR_COLLECTION_ID,
+    load_ver_override: str | None = QUERY_VALIDATOR_LOAD_VERSION_OVERRIDE,
+    user: KBaseUser = Depends(_OPT_AUTH)
+):
     store = app_state.get_storage(r)
-    ac = await store.get_collection_active(collection_id)
-    load_ver = get_load_version(ac, ID)
+    load_ver = await get_load_version(store, collection_id, ID, load_ver_override, user)
```

Comment:
can you remind a use case when an admin wants to use an overridden load_ver?

Reply:
Without an override, the data is only viewable in the API for active collections, so only one load version of the data is viewable per collection. If you load new data, you can override the load versions to see it.

I considered allowing specification of a collection version instead, but that just effectively overrides the load version as well, so I figured we might as well allow that directly so we can view data that doesn't even have an inactive collection yet.